analyzers/src/SonarAnalyzer.CSharp/Rules/TypeExaminationOnSystemType.cs

```diff
@@ -103,23 +95,22 @@ protected override void Initialize(SonarAnalysisContext context)
                 return;
             }
 
-            context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, memberCall.OperatorToken.CreateLocation(invocation),
-                MessageGetType));
+            context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, memberCall.OperatorToken.CreateLocation(invocation), MessageGetType));
         }
 
-        private static bool IsGetTypeCall(IMethodSymbol invokedMethod)
-        {
-            return invokedMethod.Name == "GetType" &&
-                !invokedMethod.IsStatic &&
-                invokedMethod.ContainingType != null &&
-                IsObjectOrType(invokedMethod.ContainingType);
-        }
+        private static bool IsException(MemberAccessExpressionSyntax memberAccess, SemanticModel semanticModel) =>
+            memberAccess.Expression is TypeOfExpressionSyntax typeOf
+            && typeOf.Type.IsKnownType(KnownType.System_Type, semanticModel);
 
-        private static bool IsObjectOrType(INamedTypeSymbol namedType)
-        {
-            return namedType.SpecialType == SpecialType.System_Object ||
-                namedType.Is(KnownType.System_Type);
-        }
+        private static bool IsGetTypeCall(ISymbol invokedMethod) =>
+            invokedMethod.Name == nameof(Type.GetType)
+            && !invokedMethod.IsStatic
+            && invokedMethod.ContainingType != null
+            && IsObjectOrType(invokedMethod.ContainingType);
+
+        private static bool IsObjectOrType(ITypeSymbol namedType) =>
+            namedType.SpecialType == SpecialType.System_Object
+            || namedType.Is(KnownType.System_Type);
 
         internal static bool IsGetTypeCall(InvocationExpressionSyntax invocation, SemanticModel semanticModel)
```

Comment:
It should be possible to simplify to something like this:
```suggestion
        internal static bool IsGetTypeCall(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
            invocation != null && semanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol && IsGetTypeCall(methodSymbol);
```
And It would be nice to consider moving the method to a helper, as other rules use this method too.